sw/device/tests/sim_dv/flash_ctrl_lc_rw_en_test.c

```diff
@@ -0,0 +1,222 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/dif/dif_keymgr.h"
+#include "sw/device/lib/dif/dif_kmac.h"
+#include "sw/device/lib/dif/dif_lc_ctrl.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_flash_ctrl_state_t flash;
+static dif_kmac_t kmac;
+static dif_keymgr_t keymgr;
+static dif_lc_ctrl_t lc;
+static dif_otp_ctrl_t otp;
```

Comment:
As far as I understood, these variables are not being used directly by any function other than the `test_main()`. So I would declare then inside `test_main()`